src/logical/chrono.rs

```diff
@@ -94,13 +121,58 @@ impl LogicalArrayType<NaiveTime> for NaiveTime {
 pub type NaiveTimeArray<const NULLABLE: bool = false, Buffer = crate::buffer::VecBuffer> =
     LogicalArray<NaiveTime, NULLABLE, Buffer, crate::offset::NA, crate::array::union::NA>;
 
+impl ArrayType<TimeDelta> for TimeDelta {
+    type Array<Buffer: BufferType, OffsetItem: OffsetElement, UnionLayout: UnionType> =
+    LogicalArray<Self, false, Buffer, OffsetItem, UnionLayout>;
+}
+
+impl ArrayType<TimeDelta> for Option<TimeDelta> {
+    type Array<Buffer: BufferType, OffsetItem: OffsetElement, UnionLayout: UnionType> =
+    LogicalArray<TimeDelta, true, Buffer, OffsetItem, UnionLayout>;
+}
+
+/// The number of nano seconds in a milli second.
+const NANOS_PER_MILLI: i64 = 1_000_000;
+/// The number of milli seconds in a second.
+const MILLI_SECONDS: i64 = 1_000;
+
+impl LogicalArrayType<TimeDelta> for TimeDelta {
+    type ArrayType = i64;
+
+    fn from_array_type(item: Self::ArrayType) -> Self {
+        let (secs, nano) = (item.div_euclid(MILLI_SECONDS),
+                            u32::try_from(item.rem_euclid(NANOS_PER_MILLI))
+                                .expect("i64 to u32 cast error"));
+        
+        Self::new(secs, nano).expect("out of range")
+    }
+
+    fn into_array_type(self) -> Self::ArrayType {
+        self.num_seconds() * NANOSECONDS + i64::from(self.subsec_nanos())
+    }
+}
+
+/// An array for [`TimeDelta`] items.
+pub type TimeDeltaArray<const NULLABLE: bool = false, Buffer = crate::buffer::VecBuffer> =
+LogicalArray<TimeDelta, NULLABLE, Buffer, crate::offset::NA, crate::array::union::NA>;
+
 #[cfg(test)]
 mod tests {
     use super::*;
     use crate::Length;
 
     #[test]
-    fn round_trip() {
+    fn round_trip_naivedate() {
```

Comment:
To make sure this test works as a regression test we should use the trait conversion methods instead of copying the implementation details here.

Reply:
Is this right?

```rs
#[test]
fn round_trip_naivedate() {
    for value in [
        NaiveDate::from_yo_opt(2024, 7).expect("out of range"),
        NaiveDate::from_yo_opt(2020, 6).expect("out of range"),
    ] {
        let days_from_ce: i32 = value.into();
        let round_tripped: NaiveDate = days_from_ce.into();
        assert_eq!(round_tripped, value);
    }
}
```

Reply:
I meant using the `from_array_type` and `.into_array_type` methods: `assert_eq!(...::from_array_type(value).into_array_type(), value)`.

Reply:
I can be a bit slow :sweat_smile:. Think I got it right now. Pls check.